src/main/kotlin/se/uulm/snowballr/backend/repository/association/ProjectPaperTableRepo.kt

```diff
@@ -168,4 +180,21 @@ class ProjectPaperTableRepo(
                 it[createdBy] = userId
             }
     }
+
+    override suspend fun getProjectProgress(projectId: UUID): Float = db.query {
+        val allPapers = ProjectPaperTable.selectAll().where { ProjectPaperTable.projectId eq projectId }.count()
+        if (allPapers == 0L) {
+            0.0f
+        } else {
+            val fullyReviewedPapers = ProjectPaperTable.selectAll().where {
+                (
+                    (ProjectPaperTable.decision eq PaperDecision.PAPER_DECISION_ACCEPTED) or
+                        (ProjectPaperTable.decision eq PaperDecision.PAPER_DECISION_DECLINED)
+                    ) and
+                    (ProjectPaperTable.projectId eq projectId)
+            }.count()
+            val progress = fullyReviewedPapers.toFloat() / allPapers
+            progress
```

Comment:
What about extracting the fully reviewed operator to a variable to increase readability and decrease unnecessary indentation?
```suggestion
            val isPaperReviewedOp = (ProjectPaperTable.decision eq PaperDecision.PAPER_DECISION_ACCEPTED) or
                    (ProjectPaperTable.decision eq PaperDecision.PAPER_DECISION_DECLINED)
            val fullyReviewedPapers = ProjectPaperTable.selectAll().where {
                isPaperReviewedOp and (ProjectPaperTable.projectId eq projectId)
            }.count()

            val progress = fullyReviewedPapers.toFloat() / allPapers
            progress
```

Reply:
This does not work as `eq` seems to be only present inside the `where` clause. I've moved it inside, which still improves readability.

Furthermore, I quote: _"Non-boolean properties shouldn't start with 'is' prefix. Actual type of isPaperReviewedOp: Op [NonBooleanPropertyPrefixedWithIs]"_. I will call it `paperReviewedOp`, even though it is a worse name.

Reply:
For the `eq` you require the `org.jetbrains.exposed.sql.SqlExpressionBuilder.eq` import. And yes, the linting error is a pain in the ass. It sadly can't be configured to ignore other types. I'm not sure what the best way to deal with this is. For the access rules, we ignored the linting error.

I'm fine with the current solution. If you would like to move the `paperReviewOp` out of the where block, you can do so. Otherwise, close this conversation.

Reply:
I think both is fine. I'd leave it, as you also don't seem to have a strong preference.